influxql/parser.go

```diff
@@ -1158,6 +1160,32 @@ func (p *Parser) parseRetentionPolicy() (name string, dfault bool, err error) {
 	return
 }
 
+// parseShowStatsStatement parses a string and returns a ShowStatsStatement.
+// This function assumes the "SHOW STATS" tokens have already been consumed.
+func (p *Parser) parseShowStatsStatement() (*ShowStatsStatement, error) {
+	stmt := &ShowStatsStatement{}
+
+	tok, pos, lit := p.scanIgnoreWhitespace()
+	if tok == EOF {
+		// No more to do.
+		return stmt, nil
+	}
+
+	// Look for the optional ON.
+	if tok == ON {
+		// Parse the host
+		ident, err := p.parseString()
+		if err != nil {
+			return nil, err
+		}
+		stmt.Host = ident
+		return stmt, err
+	}
+
+	// Anything else is an error.
+	return nil, newParseError(tokstr(tok, lit), []string{"ON"}, pos)
+}
+
```

Comment:
Can't check for `EOF` because there could be a semicolon and another statement following.  `SHOW STATS XXX` will get caught by query level parsing.  This will merge into your branch if you like this change...
https://github.com/influxdb/influxdb/pull/1915